include/pybind11/numpy.h

```diff
@@ -165,16 +165,16 @@ class dtype : public object {
 public:
     PYBIND11_OBJECT_DEFAULT(dtype, object, detail::npy_api::get().PyArrayDescr_Check_);
 
-    dtype(const buffer_info &info) {
+    explicit dtype(const buffer_info &info) {
         dtype descr(_dtype_from_pep3118()(PYBIND11_STR_TYPE(info.format)));
         m_ptr = descr.strip_padding().release().ptr();
     }
 
-    dtype(std::string format) {
+    explicit dtype(std::string format) {
         m_ptr = from_args(pybind11::str(format)).release().ptr();
     }
 
-    dtype(const char *format) : dtype(std::string(format)) { }
+    explicit dtype(const char *format) : dtype(std::string(format)) { }
```

Comment:
Ok, let's reinstate that one then. Can you include it with the upcoming flags PR?